dags/utils/hdx_dags_filler.py

```diff
@@ -0,0 +1,169 @@
+from airflow.operators.pipeline_plugin import (
+    DefaultTransformOperator,
+    HDXAdmTransformOperator,
+    HDXExtractOperator,
+)
+from pipeline_plugin.transform.default_transform import default_transform
+
+
+def fill_hdx_roads_dag(dag, config, country):
+    if config.should_process_hdx_roads(country):
+        hdx_roads_extract = HDXExtractOperator(
+            task_id=f"{country}_hdx_roads_extract",
+            hdx_address=config.get_hdx_address(country=country, hdx_type="roads"),
+            hdx_filename=config.get_hdx_dataset_name(country=country, hdx_type="roads"),
+            output_filename=config.get_cod_raw_filename(
+                country=country, datatype="roads_cod_raw"
+            ),
+            dag=dag,
+        )
+        source = "cod"
+        roads_transform = DefaultTransformOperator(
+            task_id=f"{country}_hdx_roads_transform",
+            source=source,
+            input_filename=config.get_cod_raw_filename(
+                country=country, datatype="roads_cod_raw"
+            ),
+            output_filename=config.get_cod_processed_filepath(
+                country=country, datatype="roads"
+            ),
+            crs=config.get_crs(),
+            schema_mapping=config.get_schema_mapping(
+                source=source, country=country, dataset_name="roads"
+            ),
+            transform_method=default_transform,
+            dag=dag,
+        )
+
+        hdx_roads_extract >> roads_transform
+
+
+def fill_hdx_adm_dag(dag, config, country):
```

Comment:
This is a big improvement. However, it still feels like there ought to be a way to further generalise the calls to `HDXAdmTransformOperator` for these values `adm0_transform, adm1_transform, adm2_transform`

```
# Possible generalisation
adm_params = ["adm0", "adm1", "adm2", "adm3"]
adm_transforms = []
for admx in adm_params:
	adm_transforms.append(HDXAdmTransformOperator(
		...
	))
```

Reply:
Now a JIRA issue here https://mapaction.atlassian.net/browse/DATAPIPE-95